### DISCUSSION TITLE
Windows server identity

### DISCUSSION THREAD

**Discussion by TigerMax** [strongswan/strongswan#1076] [Q&A] [answered]
Good evening.
I'am currently transforming our vpn network of seperate dedicated hosts into auto-scaling groups inside docker containers.
We've used different certificates on each machine, dedicated to each of our teams.

Now i have a single certificate tied to domain "vpn.exampledomain.com" with each container.
When connecting from android(App is amazing), i can set a host machine IP as an adress and "vpn.exampledomain.com" as server identity and everything works just fine.
But 99% of our machines are running windows, where either such option doesn't exist or I'am simply blind. (Using native windows VPN service)

Changing leftid=<serverip> didn't help.

Is there a way to skip this check or to set such option on windows machine or  to change some configs to achieve my goal?

Any help or hints in the right direction would be much appreciated.

ipsec.conf

```
config setup
  uniqueids=no

conn %default
  keyexchange=ikev2
  ikelifetime=60m
  keylife=20m
  rekeymargin=3m
  keyingtries=1
  rekey=no
  ike=chacha20poly1305-prfsha256-newhope128,chacha20poly1305-prfsha256-ecp256,aes128gcm16-prfsha256-ecp256,aes256-sha256-modp2048,aes256-sha256-modp1024!
  esp=chacha20poly1305-newhope128,chacha20poly1305-ecp256,aes128gcm16-ecp256,aes256-sha256-modp2048,aes256-sha256,aes256-sha1!
  dpdaction=clear
  dpddelay=120s
  auto=add

conn roadwarrior
  left=%any
  leftid=@vpn.exampledomain.com # Its fake, ofc
  leftauth=pubkey
  leftcert=serverCert.pem
  leftsendcert=always
  leftsubnet=0.0.0.0/0,::/0
  leftupdown=/etc/ipsec.d/firewall.updown
  right=%any
  rightauth=pubkey
  rightsourceip=192.168.12.0/24,fdef:a51d:f888::/112
  rightdns=8.8.8.8,2001:4860:4860::8888

conn roadwarrior-eap
  also=roadwarrior
  rightauth=eap-dynamic
  eap_identity=%any

conn roadwarrior-pubkey-eap
  also=roadwarrior
  rightauth2=eap-dynamic
  eap_identity=%any

conn client
  left=%defaultroute
  leftauth=eap
  leftsourceip=%config4,%config6
  rightauth=pubkey
  rightsubnet=0.0.0.0/0,::/0
  eap_identity=%identity

conn home
  also=client
  leftid=carol
  right=moon.strongswan.org
  rightid=@moon.strongswan.org

```

**Comment by tobiasbrunner** [comment, marked as answer]
As far as I know, the Windows client does not allow configuring a specific server identity (not even via PowerShell). So it will enforce the configured IP address or host name as identity.

Therefore, if you want to use a single certificate, you'll have to add all IPs (and/or host names) the clients will use as subjectAltName extensions to it. I'm not sure if the Windows client requires that the IKE identity returned by the server matches the identity it assumed, or if it's enough if that identity is simply contained as subjectAltName in the server certificate. Depending on that you can either keep `leftid` the same, or you'll have to change it for each server instance to the correct IP/host name the clients will use when connecting.

**Reply by TigerMax** [reply]
Thank you for your response!
Unfortunately, adding IP's in advance is not possible,
because we plan to have ~30 servers with dynamic IP adresses with max lifetime of 12 hours.

Is certeficate-less authentication(like PSK) is my only option?

Thanks you again!

**Reply by tobiasbrunner** [reply]
So on Android you have to change the IP address all the time? Wouldn't FQDNs, which you could add as SAN, with changing A records be easier to manage?

> Is certeficate-less authentication(like PSK) is my only option?

Note that neither our Android client nor the Windows IKEv2 client support PSK authentication.

**Reply by TigerMax** [reply]
Thank you for your reply!
> So on Android you have to change the IP address all the time?

I mostly use android for ease of troubleshooting. Logs are easily acessible.
VPN servers are auto-scaling in the cloud. I have written windows application that manages certificates, creates vpn connection, parses server IP from our backend, applies split-tunneling settings e.t.c.
Constantly changing IP's aren't a problem. 
I wanted to have a single certeficate just for ease of use. 
Will have to make them during init. Its safer that way anyway.

> Wouldn't FQDNs, which you could add as SAN, with changing A records be easier to manage?

I honestly didn't know such possibility existed. Thank you. I will look into it.


> Note that neither our Android client nor the Windows IKEv2 client support PSK authentication.

Yeah, learned that already. That is unfortunate(
